### PR TITLE
Fixed a regression with DFLAGS environment variable and cached generated unittest

### PR DESCRIPTION
**Pull request by foerdi**
I fixed a regression that @Geod24 found: https://github.com/dlang/dub/pull/2005#issuecomment-753707610

Should I or @Geod24 open an issue and link it with this PR?

### CODE REVIEW

File: source/dub/dub.d [hunk 1]
```diff
@@ -701,10 +701,10 @@ class Dub {
 			mkdirRecurse(mainfile.parentPath.toNativeString());
 
 			bool regenerateMainFile = settings.force || !existsFile(mainfile);
-
+			auto escapedMainFile = mainfile.toNativeString().replace("$", "$$");
```
Comment (Geod24) [inline source/dub/dub.d:704]: https://dlang.org/phobos/std_process.html#.escapeShellFileName

File: test/cache-generated-test-config.sh [hunk 2]
```diff
@@ -4,6 +4,7 @@
 cd ${CURR_DIR}/cache-generated-test-config
 rm -rf .dub
 
+## defult test
```
Comment (Geod24) [inline test/cache-generated-test-config.sh:7]:
```suggestion
## default test
```

**Comment by Geod24** [comment]
Can you also add a test with a non-empty DFLAGS ?

**Comment by Geod24** [comment]
Also please target stable

File: source/dub/dub.d (hunk 1 above)
Comment (foerdi) [reply source/dub/dub.d:704]: unfortunately `escapeShellFileName` is not working, I got this Error `Error: unrecognized file extension d'`

**Comment by foerdi** [comment]
> Can you also add a test with a non-empty DFLAGS ?
> Also please target stable

sure

File: test/cache-generated-test-config.sh (hunk 2 above)
Comment (foerdi) [reply test/cache-generated-test-config.sh]: thank you

**Review by Geod24 — APPROVED** [review]
